src/Core/Components/TextArea/FluentTextArea.razor.cs

```diff
@@ -0,0 +1,135 @@
+// ------------------------------------------------------------------------
+// MIT License - Copyright (c) Microsoft Corporation. All rights reserved.
+// ------------------------------------------------------------------------
+
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Web;
+using Microsoft.JSInterop;
+
+namespace Microsoft.FluentUI.AspNetCore.Components;
+
+/// <summary>
+/// A textarea component that allows users to enter and edit a single line of text.
+/// </summary>
+public partial class FluentTextArea : FluentInputImmediateBase<string?>, IFluentComponentElementBase
+{
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="FluentTextInput"/> class.
+    /// </summary>
+    public FluentTextArea()
+    {
+        // Default conditions for the message
+        MessageCondition = (field) =>
+        {
+            field.MessageIcon = FluentStatus.ErrorIcon;
+            field.Message = Localizer[Localization.LanguageResource.TextInput_RequiredMessage];
+
+            return FocusLost &&
+                   (Required ?? false)
+                   && !(Disabled ?? false)
+                   && !ReadOnly
+                   && string.IsNullOrEmpty(CurrentValueAsString);
+        };
+    }
+
+    /// <inheritdoc cref="IFluentComponentElementBase.Element" />
+    [Parameter]
+    public ElementReference Element { get; set; }
+
+    /// <summary>
+    /// Gets or sets the visual appearance.
+    /// </summary>
+    [Parameter]
+    public TextAreaAppearance? Appearance { get; set; }
+
+    /// <summary>
+    /// Gets or sets the short hint displayed in the textarea before the user enters a value.
+    /// </summary>
+    [Parameter]
+    public string? Placeholder { get; set; }
+
+    /// <summary>
+    /// Gets or sets the maximum number of characters allowed in the textarea
+    /// </summary>
+    [Parameter]
+    public int? MaxLength { get; set; }
+
+    /// <summary>
+    /// Gets or sets the minimum number of characters allowed in the textarea
+    /// </summary>
+    [Parameter]
+    public int? MinLength { get; set; }
+
+    /// <summary>
+    /// Specifies whether a form or an textarea field should have autocomplete "on" or "off" or another value.
+    /// An Id value must be set to use this property.
+    /// </summary>
+    [Parameter]
+    public string? AutoComplete { get; set; }
+
+    /// <summary>
+    /// Whether the element’s height should be automatically changed based on the content.
+    /// </summary>
+    [Parameter]
+    public bool? AutoResize { get; set; }
+
+    /// <summary>
+    /// Whether the element displays a visual box shadow
+    /// </summary>
+    [Parameter]
+    public bool? DisplayShadow { get; set; }
```

Comment:
Maybe the `Appearance` could be contains these items, to be coherent with React (and remove `DisplayShadow` attribute):
- Outline
- FilledLighter
- FilledDarker
- FilledLighterShadow
- FilledDarkerShadow

https://react.fluentui.dev/?path=/docs/components-textarea--docs